### ISSUE TITLE
Positional options lost in 1.0

### ISSUE THREAD

**Issue by bilherron** [sir-dunxalot/ember-tooltips#82]
As I was up upgrading this (fantastic!) addon, it looks like we have lost some granularity in positioning the tooltip (ie top-left, top-right, bottom-left, bottom-right). I think ember-tether can support these, and I'm happy to take a crack at implementing it in ember-tooltips but I wanted to check in first on your thoughts @sir-dunxalot. Thanks!

**Comment by sir-dunxalot** [comment]
Agreed. I'd love to put them back in. They were removed for simplicity to get 1.0 out the door and I fully support reimplementing them.

We will need to make sure that both side and [attachment](https://github.com/sir-dunxalot/ember-tooltips/blob/master/addon/components/tooltip-on-element.js#L78) function correctly with regards to what ember-tether expects.

Also, [position class](https://github.com/sir-dunxalot/ember-tooltips/blob/master/addon/components/tooltip-on-element.js#L124) should support these new multi-word positions.

Also, see [this bit of code](https://github.com/sir-dunxalot/ember-tooltips/blob/master/addon/components/tooltip-on-element.js#L305-L335), which is a little hacky but presents options in a format that ember-tether expects.

Tests can be added to [the side integration test](https://github.com/sir-dunxalot/ember-tooltips/blob/master/tests/integration/components/side-test.js).

**Comment by sir-dunxalot** [comment]
I also wonder if `position` is a better property name than `side`?

**Comment by sir-dunxalot** [comment]
Hey @bilherron, I don't suppose you've been able to take a crack at this?

**Comment by bilherron** [comment]
Haven't had a chance yet, sorry! Still on my radar, though.

**Comment by renatocarvalho** [comment]
To me, this change was a big regression. Darsain tooltip is great because it calculates everything automatically. Now everything in breaking on my app. I like the idea of keeping things simple, but in order to make tooltips just work, it needs a lot of work. And that's what Darsain Tooltip did.

@sir-dunxalot Thanks for all your hard work though.

**Comment by sir-dunxalot** [comment]
Thanks for the feedback @renatocarvalho! It's been a little rocky but it will work out in the end. :)

**Comment by a15n** [comment]
@jcward10 has https://github.com/sir-dunxalot/ember-tooltips/pull/103 that would introduce more positional options. His PR introduces a `defaultPosition` which acts like a secondary position (if I understand correctly)...

```
{{#tooltip-on-component side='right' defaultPosition='top'}}
  template block text
{{/tooltip-on-component}}
```

While this PR introduces more granularity it still doesn't support all of the functionality built into [tether](http://tether.io/) itself. From my POV, the `side` parameter is a useful option that covers the vast majority of use cases. Under the hood the `side` param is used to calculate `attachment` and `targetAttachment`. However, if you want to position the tooltip in any custom way you can simply override these parameters (we should add this to the documentation).

```
<div class="box">
	hover on me
	{{#tooltip-on-element
		attachment="top left"
		targetAttachment="top right"
	}}
		tooltip text here
	{{/tooltip-on-element}}
</div>
```

<img width="273" alt="screen shot 2016-10-19 at 3 25 37 pm" src="https://cloud.githubusercontent.com/assets/7050871/19539801/510540c6-9610-11e6-8f04-94d210f96d6d.png">


Should we push to add additional positional params or just document the existing functionality of the underlying tether component?

**Comment by sir-dunxalot** [comment]
The reason `side` was originally implemented was because `attachment` and `targetAttachment` were not very intuitive. It's `targetAttachment` that is really pertinent to the tooltip developer. Developers shouldn't have to set two params to move the tooltip.

I wonder if `side` could take two options like `left` and `right`, as it does now, but also new options like `top left` and `bottom right` where the first word is the side and the second word is the position on the side. We could even split this into two properties: `side`, which functions as it does not and a param like `positionOnSide`, which controls the extra positioning of the tooltip.

**Comment by jcward10** [comment]
Doesn't side take more than two options ? `top` `right` `bottom` `left`. It would be nice to add positioning for the tooltip arrow. So when we change the tool tips `attachment` and `targetAttachment` we can move the arrow to the top of the tooltip.

**Comment by andrew-zenefits** [comment]
@jcward10 right now `side` only accepts a one-word string. I think @sir-dunxalot's recommendation to approach with a two-word `side`  may work, like "top left", where the first word is the `side` and the `second`, if provided, is `positionOnSide`. 

Your PR essentially does this, but with two variables and I used it as a POC for all the images below. I think that if we want to support a modifying variable we will need to provide an opinionated positioning framework on where to place the tooltips under all scenarios. This will likely mean applying a modifying variable to targetAttachment as well.

Jared can you please explore further how this would work under the following scenarios....

* tall element & short tooltip on right (see first pic for current behavior in https://github.com/sir-dunxalot/ember-tooltips/pull/103)
* short element & tall tooltip on right (see second pic for current behavior in https://github.com/sir-dunxalot/ember-tooltips/pull/103)
* wide element & narrow tooltip on top
* narrow element & wide tooltip on top

### tall element & short tooltip on right
![screen shot 2016-10-21 at 6 00 07 pm](https://cloud.githubusercontent.com/assets/8782940/19615706/453dac34-97b8-11e6-9bb8-44c1329f2ba1.png)


### short element & tall tooltip on right
![screen shot 2016-10-21 at 6 01 06 pm](https://cloud.githubusercontent.com/assets/8782940/19615710/5bd90178-97b8-11e6-8371-834b148dab59.png)

PS: Here is how it looks when you use the set the `attachment` and `targetAttachment` individually. I understand that they are not the most intuitive but they are the most powerful and expressive (if we want to publicize these as options we'll need to fix the arrow placement)

![screen shot 2016-10-21 at 6 03 10 pm](https://cloud.githubusercontent.com/assets/8782940/19615734/d42a4416-97b8-11e6-944a-2279e7cdeaa7.png)

**Comment by a15n** [comment]
@jcward10 right now `side` only accepts a one-word string. I think @sir-dunxalot's recommendation to approach with a two-word `side`  may work, like "top left", where the first word is the `side` and the `second`, if provided, is `positionOnSide`. 

Your PR essentially does this, but with two variables and I used it as a POC for all the images below. I think that if we want to support a modifying variable we will need to provide an opinionated positioning framework on where to place the tooltips under all scenarios. This will likely mean applying a modifying variable to targetAttachment as well.

Jared can you please explore further how this would work under the following scenarios....

* tall element & short tooltip on right (see first pic for current behavior in https://github.com/sir-dunxalot/ember-tooltips/pull/103)
* short element & tall tooltip on right (see second pic for current behavior in https://github.com/sir-dunxalot/ember-tooltips/pull/103)
* wide element & narrow tooltip on top
* narrow element & wide tooltip on top

### tall element & short tooltip on right
![screen shot 2016-10-21 at 6 00 07 pm](https://cloud.githubusercontent.com/assets/8782940/19615706/453dac34-97b8-11e6-9bb8-44c1329f2ba1.png)


### short element & tall tooltip on right
![screen shot 2016-10-21 at 6 01 06 pm](https://cloud.githubusercontent.com/assets/8782940/19615710/5bd90178-97b8-11e6-8371-834b148dab59.png)

PS: Here is how it looks when you use the set the `attachment` and `targetAttachment` individually. I understand that they are not the most intuitive but they are the most powerful and expressive (if we want to publicize these as options we'll need to fix the arrow placement)

![screen shot 2016-10-21 at 6 03 10 pm](https://cloud.githubusercontent.com/assets/8782940/19615734/d42a4416-97b8-11e6-944a-2279e7cdeaa7.png)

**Comment by jcward10** [comment]
Looks good! Let me know how I can help. If someone wants to work on refactoring my PR I'm available to pair almost any time.

**Comment by humanchimp** [comment]
> Should we push to add additional positional params or just document the existing functionality of the underlying tether component?

Learning about those underlying attributes was all that I needed to address all of the use-cases found in our moderately large app, as we ported our popovers and tooltips from [this library](https://github.com/lytics/ember-components).

It seems like additional documentation is all that is required.
